tests/azure_kv_test.py

```diff
@@ -0,0 +1,95 @@
+"""Tests Azure Key Vault credential plugin."""
+
+# ruff: noqa: S106 UP045
+from typing import Optional, TypedDict
+
+import pytest
+
+from azure.identity import CredentialUnavailableError
+from azure.keyvault.secrets import (
+    KeyVaultSecret,
+    SecretClient,
+    SecretProperties,
+)
+
+from awx_plugins.credentials import azure_kv
+
+
+class _FakeSecretClient(SecretClient):
+    def get_secret(
+        self: '_FakeSecretClient',
+        name: str,
+        version: Optional[str] = None,
+        **kwargs: str,
+    ) -> KeyVaultSecret:
+        props = SecretProperties()
+        return KeyVaultSecret(properties=props, value='test-secret')
+
+
+class AzureCredentials(TypedDict):
+    """Schema for a collection of plugin input fields."""
+
+    url: str
+    client: str
+    secret: str
+    tenant: str
+    secret_field: str
+    secret_version: str
+
+
+def test_azure_kv_invalid_env() -> None:
+    """Test running outside of Azure raises error."""
+    with pytest.raises(
+        CredentialUnavailableError,
```

Comment:
If this is happening, something is not being handled right in the implementation. I'd expect a `RuntimeError` and not underlying errors leaking into this layer. This is probably a bug somewhere that has to be fixed before merging.

Reply:
We have to figure out why it doesn't hit https://app.codecov.io/gh/ansible/awx-plugins/commit/94f7e3611fd08fcec49e22d81d6f7618ab8567fc/blob/src/awx_plugins/credentials/azure_kv.py?flags%5B0%5D=pytest&dropdown=coverage#L153